### ISSUE TITLE
Search interfaces, front ports, rear ports, ... in hierarchical structured locations

### ISSUE THREAD

**Issue by baschdello** [netbox-community/netbox#8388]
### NetBox version

v3.1.5

### Feature type

Change to existing functionality

### Proposed functionality

There exists site groups and locations which can be hierarchical ordered. For example a location tree can be:
block2
| --- rackrow3
| --- rackrow4
At the device search it is possible to select the parent location "block2" and netbox is listing all devices from racks appended to the child locations "rackrow3" and "rackrow4".
It will be useful to have the same filtering method at the Interface search. However if I want to list all interfaces located in parent location "block2" the result is empty. They will only be listed if selecting the child location "rackrow3" or "rackrow4" in location filter directly.
The same filtering functionality should be present at api calls.

### Use case

This functionality adds some consistencies regarding the filtering function in UI because this filtering logic is already present at device filtering. Adding this filtering functionality at interfaces, front ports, rear ports, console ports, console server ports, power ports, power outlets will help to get a consistent filtering logic.

Another benefit at using api calls: it can simplify scripted cabling. If the interface of partner device (located in rackrow3) can not be found in the same row it is possible to jump to parent location "block2" and search the partner interface there (it is located in rackrow4). The benefit will be that there is no need to do a separate search for every child location.

**Comment by baschdello** [comment]
#8370

**Comment by DanSheps** [comment]
To clarify, currently the interface search allows you to filter the location/site/etc but doesn't let you search for the parent location and show the child locations.  Example:

Region: United States -> New York -> New York City
Site: Test
Location: Room: 250 -> Cage 10 -> Row 1
Rack: 2342
Device: Core Router
Interface 10

You can search for "Row 1", New York City, but not New York.

**Comment by baschdello** [comment]
I tried it again because I did not use region filter before: if I use regions as filter options at the interface search it works well for parent and child regions. In your example "New York" will return interfaces and "New York City" will return interfaces. So filter option "region" seems to be correctly implemented here.
But using locations as filter options it works only with child location. In your example "Row 1" will result interfaces. But using the location "Cage 10" the result is empty.
